config/index.js

```diff
@@ -31,9 +31,24 @@ function applyOverrides (config, allOverrides, overrides) {
   return config
 }
 
-function getOverrideFilePath (overridePath) {
+function getOverrideFilePath (config, overridePath) {
   const serviceEnv = BaseServiceConfig.getServiceEnvironment()
-  const env = (serviceEnv && serviceEnv.toLowerCase()) || 'localhost'
+  let env
+
+  if (serviceEnv) {
```

Comment:
Some extra safety checks to ensure that an environment that is provided is valid in the `Environment` enum model.